src/main/java/core/basesyntax/stategy/handler/PurchaseOperation.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.stategy.handler;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void process(FruitTransaction transaction) {
+        Integer fruitQuantity = Storage.fruits.get(transaction.getFruitName());
+        int purchaseQuantity = transaction.getQuantity();
+        if (purchaseQuantity > 0 && purchaseQuantity <= fruitQuantity) {
+            int updateQuantity = fruitQuantity - purchaseQuantity;
+            Storage.fruits.put(transaction.getFruitName(), updateQuantity);
+        } else {
+            throw new IllegalArgumentException("Invalid quantity: " + purchaseQuantity);
```

Comment:
The condition 'purchaseQuantity > 0' is not necessary since the quantity should always be positive. Also, there should be a check to ensure that 'fruitQuantity' is not null before comparing it to 'purchaseQuantity' to avoid a NullPointerException.